pyvista/utilities/misc.py

```diff
@@ -163,3 +163,20 @@ def set_pickle_format(format: str):
             f'Unsupported pickle format `{format}`. Valid options are `{"`, `".join(supported)}`.'
         )
     pyvista.PICKLE_FORMAT = format
+
+
+def no_new_attr(cls):
+    """Override __setattr__ to not permit new attributes."""
+
+    def __setattr__(self, name, value):
+        """Do not allow setting attributes."""
+        if hasattr(self, name):
+            object.__setattr__(self, name, value)
+        else:
+            raise AttributeError(
+                f'Attribute "{name}" does not exist and cannot be added to type '
+                f'{self.__class__.__name__}'
+            )
+
+    setattr(cls, '__setattr__', __setattr__)
+    return cls
```

Comment:
Wow is this nice (and needed in Python std libs)

Reply:
I don't have context here so sorry if this is a dumb question, but why can't we use `__slots__`?